picard/formats/__init__.py

```diff
@@ -49,6 +71,8 @@ def open(filename):
         format = _extensions[ext]
     except KeyError:
         return None
```

Comment:
I am facing a bit of a problem. Apparently the else block is executing every time anything other than a KeyError is encountered.

Now if I place the guess_format function under the except clause it causes an error later in the thread when the file's _load method is called and I can't do anything about the format then.

How should I proceed?